omnilingo/main.py

```diff
@@ -0,0 +1,52 @@
+"""OmniLingo CLI Interface."""
+
+from pathlib import Path
+
+import click
+
+import omnilingo.cache as _cache
+import omnilingo.server as server
+
+
+@click.group()
+@click.option(
+    "--cache-path",
+    type=lambda x: Path(click.Path(exists=True, file_okay=False, writable=True)(x)),
+    default=_cache.DEFAULT_PATH,
+    help="Path to cache directory.",
+)
+@click.pass_context
+def main(ctx: click.Context, cache_path: Path) -> None:
+    """Omnilingo Control Commands."""
+    ctx.obj = cache_path
+
+
+@main.command()
+@click.option(
+    "--port",
+    type=click.IntRange(0, 65535),
```

Comment:
Do we want to allow people to run this on ports < 1000 ? e.g. should we give a warning if running as root?

Reply:
You will get an error if you try to bind a privileged port as non-root so that's already handled.  We don't need to add any processing for things like that.